tutorials/01_Basic_QA_Pipeline.ipynb

```diff
@@ -224,188 +154,149 @@
-    "\n",
-    "# We now have a list of dictionaries that we can write to our document store.\n",
-    "# If your texts come from a different source (e.g. a DB), you can of course skip convert_files_to_dicts() and create the dictionaries yourself.\n",
-    "# The default format here is:\n",
-    "# {\n",
-    "#    'content': \"<DOCUMENT_TEXT_HERE>\",\n",
-    "#    'meta': {'name': \"<DOCUMENT_NAME_HERE>\", ...}\n",
-    "# }\n",
-    "# (Optionally: you can also add more key-value-pairs here, that will be indexed as fields in Elasticsearch and\n",
-    "# can be accessed later for filtering or shown in the responses of the Pipeline)\n",
-    "\n",
-    "# Let's have a look at the first 3 entries:\n",
-    "print(docs[:3])\n",
     "\n",
-    "# Now, let's write the dicts containing documents to our DB.\n",
-    "document_store.write_documents(docs)"
-   ]
+    "fetch_archive_from_http(\n",
+    "    url=\"https://s3.eu-central-1.amazonaws.com/deepset.ai-farm-qa/datasets/documents/wiki_gameofthrones_txt1.zip\",\n",
+    "    output_dir=doc_dir\n",
+    ")"
+   ],
+   "metadata": {
+    "collapsed": false,
+    "pycharm": {
+     "is_executing": true
+    }
+   }
   },
   {
    "cell_type": "markdown",
-   "metadata": {},
    "source": [
-    "## Initialize Retriever, Reader & Pipeline\n",
-    "\n",
-    "### Retriever\n",
-    "\n",
-    "Retrievers help narrowing down the scope for the Reader to smaller units of text where a given question could be answered.\n",
-    "They use some simple but fast algorithm.\n",
-    "\n",
-    "**Here:** We use Elasticsearch's default BM25 algorithm\n",
-    "\n",
-    "**Alternatives:**\n",
-    "\n",
-    "- Customize the `BM25Retriever`with custom queries (e.g. boosting) and filters\n",
-    "- Use `TfidfRetriever` in combination with a SQL or InMemory Document store for simple prototyping and debugging\n",
-    "- Use `EmbeddingRetriever` to find candidate documents based on the similarity of embeddings (e.g. created via Sentence-BERT)\n",
-    "- Use `DensePassageRetriever` to use different embedding models for passage and query (see Tutorial 6)"
-   ]
+    "2. Convert the files you just downloaded into Haystack [Document objects](https://docs.haystack.deepset.ai/docs/documents_answers_labels#document) to write them into the DocumentStore. Apply the `clean_wiki_text` cleaning function to the text."
+   ],
+   "metadata": {
+    "collapsed": false
+   }
   },
   {
    "cell_type": "code",
    "execution_count": null,
    "metadata": {},
    "outputs": [],
    "source": [
-    "from haystack.nodes import BM25Retriever\n",
-    "\n",
-    "retriever = BM25Retriever(document_store=document_store)"
+    "from haystack.utils import clean_wiki_text, convert_files_to_docs\n",
+    "docs = convert_files_to_docs(\n",
+    "    dir_path=doc_dir,\n",
+    "    clean_func=clean_wiki_text,\n",
+    "    split_paragraphs=True\n",
+    ")"
```

Comment:
I've always had issues with the way Tut1 hides the entire concept of indexing behind `convert_files_to_docs`. This might be my chance to change it :grin: 

The problem is that `convert_files_to_docs` sells the idea that indexing is not done with pipelines. I've had contributors trying to make this function a huge monster that supports Tika and other features because he was tricked into thinking that this is the way to index files (https://github.com/deepset-ai/haystack/pull/3274). We can prevent this from happening.

I believe we should make an effort and rather show users how to make a simple, small indexing pipeline here instead of using `convert_files_to_docs`. If that implies some changes in Haystack to make this step dead simple, I'm happy to make them.

`</rant>` :smile:

Reply:
Ahhh that's interesting. Ok I see this being tied to the issue with the ready made pipelines too because they're both about abstracting away pipelines. It's definitely not great that some users are not aware of indexing pipelines. There aren't enough examples of it in our tutorials and we should do something about that. 

So I see that having proper pipeline initialisations would be much more informative from the perspective of a power user. But I still have the feeling that the abstracted `convert_files_to_docs` and `ExtractiveQA` are a good simplification for less experienced users in a Tutorial 1. I will try and verify this by asking internally or within our community.

How would you feel if we kept these abstractions in tutorial 1 but stop using them from Tutorial 2 onwards?

Reply:
I think I'm ok keeping `ExtractiveQAPipeline` for now, but I'm still a bit concerned for `convert_files_to_docs`.

If I were to introduce a `TextIndexingPipeline` into the standard pipelines to replace `convert_files_to_docs`, would that help? It's really trivial to do. At least it gets across the idea that indexing is done with pipelines, which is my biggest issues with this situation.

Reply:
Yup sure I think that's a good compromise!

Reply:
Small suggestion here which might help. Why not add a very compact beginner tutorial about Indexing Documents? Then this section can refer to that tutorial even. It could introduce the concept of indexing, refer to relevant sections in docs etc? I think it might go quite well with the new tutorial structure too. But up to you if you think this is overkill. I like the `TextIndexingPipeline` in any case! I think that would be super cool!!

Reply:
I added a [task ](https://app.zenhub.com/workspaces/documentation-team-62261307d498240018e81760/issues/deepset-ai/documentation-team/298) for that @TuanaCelik .